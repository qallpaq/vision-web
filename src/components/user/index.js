import UserContainer from "./userContainer"

export default UserContainer