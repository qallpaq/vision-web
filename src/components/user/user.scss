@import "src/scss/constants";

.user {
  &__page {
    margin-top: $header-height + 100px;
    display: flex;
  }

  &__left,
  &__right {
    width: 50%;
    height: 300px;
  }

  &__img-wrapper,
  &__img-inner {
    height: 100%;
    width: 100%;
  }

  &-content {
    font-size: 25px;
  }
}

@media screen and (max-width: 900px) {
  .user__page {
    display: block;
  }
  .user__left,
  .user__right {
    width: 100%;
    height: 300px;
  }
}
