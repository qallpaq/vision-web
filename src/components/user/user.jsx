import React from 'react'
import {createUserInfo, toNormalCase} from "../../helpers"
import './user.scss'


const UserContent = data => {
    return (
        <ul className='user-content'>
            {
                Object.keys(data).map(item => {
                    return (
                        <li key={data[item] + item} className='user-content__item'>
                            {toNormalCase(item)}: {data[item]}
                        </li>
                    )
                })
            }
        </ul>
    )
}

export const User = ({user}) => {
    return (
        <div className='container'>
            <div className="user__page">

                <div className='user__left'>
                    <div className='user__img-wrapper'>
                        <img className='user__img-inner' src={user.avatar} alt="avatar"/>
                    </div>
                </div>

                <div className='user__right'>
                    <ul className='user-content'>
                        {UserContent(createUserInfo(user))}
                    </ul>
                </div>

            </div>
        </div>
    )
}
