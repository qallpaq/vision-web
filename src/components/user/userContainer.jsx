import React, {useEffect} from 'react'
import {connect} from "react-redux"
import {Redirect} from "react-router"
import {getUser} from "../../redux/thunk"
import {User} from "./user"
import Spinner from "../spinner"
import {resIsAuth, resIsFetching, resUser} from "../../redux/reselect"


const UserContainer = ({user, isAuth, getUser, isFetching}) => {

    useEffect(() => {
        getUser()
    }, [isAuth])

    if (isFetching) return <Spinner/>

    if (isAuth === false) return <Redirect to={'/login'}/>

    return <User user={user}/>
}

const mapStateToProps = state => {
    return {
        user: resUser(state),
        isAuth: resIsAuth(state),
        isFetching: resIsFetching(state)
    }
}

export default connect(mapStateToProps, {getUser})(UserContainer)
