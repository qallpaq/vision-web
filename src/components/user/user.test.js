import React from 'react'
import {mount} from "enzyme"
import {User} from "./user"


const user = {
    avatar: 'asd'
}

describe('User', () => {
    const wrapper = mount(<User user={user}/>)

    it('snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('should contain .user-content', () => {
        const result = wrapper.find('.user-content')
        expect(result).toHaveLength(1)
    })

    it('should return img src === user.avatar', () => {
        const result = wrapper.find('img')
        expect(result.prop("src")).toEqual(user.avatar)
    })
})
