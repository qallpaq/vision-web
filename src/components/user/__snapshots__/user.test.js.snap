// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User snapshot 1`] = `
<User
  user={
    Object {
      "avatar": "asd",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="user__page"
    >
      <div
        className="user__left"
      >
        <div
          className="user__img-wrapper"
        >
          <img
            alt="avatar"
            className="user__img-inner"
            src="asd"
          />
        </div>
      </div>
      <div
        className="user__right"
      >
        <ul
          className="user-content"
        />
      </div>
    </div>
  </div>
</User>
`;
